feature/shop-api

KIRBY-PARK opened the pull request:
가게 생성 및 수정 API
- 가게 오픈시간 openingHours, 가게 마감시간 closingHours 추가
- 가게 최소 주문 금액 설정 추가
- 사장 권한 확인 및 최대 3개 생성 예외 추가

가게 조회 API
- 가게명으로 다건 조회
- 가게 단건 조회 시 등록된 메뉴 목록 조회

가게 삭제 API
- 가게 상태만 폐업 상태로 변경
- 가게 삭제 시 추가 등록 가능

KIRBY-PARK committed:
feat: Shop CRUD feature implementation
가게 CRUD 기본 기능 및 요구사항, 예외처리 적용 완료하였습니다.

KIRBY-PARK committed:
docs : Service message in Korean
- 서비스의 메세지만 영어로 해놓았어서 통일이 안되는 것 같아 한글화했습니다.

saintym commented:
develo branch 가 아니라 main 브랜치에 요청하셨습니다!!
다시 요청 부탁드릴게요

saintym commented on src/main/java/com/example/shop/controller/ShopController.java:
```diff
@@ -0,0 +1,55 @@
+    private ShopService shopService;
+
+    // 가게 단건 조회
+    @GetMapping("{id}")
```
```java
@GetMapping("/{id}")
```
{id} 앞에 / 가 있는 편이 좋아보입니다.

KIRBY-PARK replied:
넵 ! 알려주셔서 감사합니다 ! 수정 완료했습니다 !

asdd2557 reviewed (approved):
lgtm

saintym commented on src/main/java/com/example/shop/service/ShopService.java:
```diff
@@ -0,0 +1,88 @@
+    public void delete(int id) {
+        Shop shop = shopRepository.findActiveById(id);
+        if (shop == null) {
+            throw new IllegalArgumentException("'" +id + "' 가게를 찾을 수 없습니다.");
```
각 에러들에는 ID 값과 같이 shop name 등을 보내준다면 이해하기 쉬울거라 생각합니다.

KIRBY-PARK replied:
오! 말씀주신대로 변경해보았습니다!

이 부분은 네번째 커밋에서 확인해보실 수 있습니다!
괜찮게 수정되었는지 한 번 더 코드리뷰 부탁드릴게요...!

saintym commented on src/main/java/com/example/shop/service/ShopService.java:
```diff
@@ -0,0 +1,88 @@
+                .orElseThrow(() -> new IllegalArgumentException("사용자를 찾을 수 없습니다." + AuthUtil.getId()));
+
+        // is_owner 여부 확인 및 가게 개수 제한
+        if (!user.getIsOwner()) {
```
user 에는 isOwner 가 존재하지 않습니다.

KIRBY-PARK replied:
엇.. 이 부분은 수정하기에 제가 이해가 안되서 조금 어렵습니다..!

User 엔티티에 private Boolean isOwner; 이 부분이 있는데 isOwner를 쓸 수 없...없는걸까요..?!

saintym replied:
지금 isOwner 는 삭제되고 Role 만 있는걸로 압니다

KIRBY-PARK replied:
수정 완료되었습니다!

7번째 커밋에서 확인 부탁드릴게요 !

saintym commented on src/main/java/com/example/shop/repository/ShopRepository.java:
@Query 를 통한 query 작성문보다, QueryDSL 을 이용해보는건 어떨까요?
현재 저희 프로젝트는 queryDSL 사용을 권장하고 있습니다.

KIRBY-PARK replied:
앗.. 오늘 여쭤보려고 했는데 사실 제가 QueryDSL을 사용할 줄 몰라서 우선 @query를 사용해놓았던 것이었는데 작업해놓으신 것 따라하면서 도전해보겠습니다 !

saintym replied:
언제든 말씀주시면 도와드리겠읍니다

KIRBY-PARK replied:
크으으으으으으으으으으으으
혁규님 덕에 해결 완료입니다 !!!!!!!!!

saintym commented on src/main/java/com/example/shop/entity/Shop.java:
```diff
@@ -1,39 +1,60 @@
-    private LocalDateTime updatedAt;
+    // 오픈 시간
+    @Column(nullable = false)
+    private String openingHours;
```
openedAt 이 컨벤션에 맞다 생각됩니다.

KIRBY-PARK replied:
이 부분은 수정 완료되었습니다 !

세번째 커밋 [ chore: Change member variable name ] 에 업데이트되었습니다 👍

saintym commented on src/main/java/com/example/shop/entity/Shop.java:
```diff
@@ -1,39 +1,60 @@
+    // 마감 시간
     @Column(nullable = false)
-    private Boolean isDeleted = false;
+    private String closingHours;
```
위와 동일하게, closedAt 이 맞다 생각됩니다.

KIRBY-PARK replied:
이 부분은 수정 완료되었습니다 !

세번째 커밋 [ chore: Change member variable name ] 에 업데이트되었습니다 👍

saintym commented on src/main/java/com/example/shop/entity/Shop.java:
```diff
@@ -1,39 +1,60 @@
+
+    // 최소 주문 금액
+    @Column(nullable = false)
+    private double minOrderAmount;
```
Amount 는 양을 뜻하는 단어라 "갯수" 라는 의미가 더 강하다 생각합니다.
minOrderPrice 를 추천하고 싶습니다.

KIRBY-PARK replied:
Price로 추천해주셔서 감사합니다 !
이 부분은 minOrderPrice로 수정 완료되었습니다 !

세번째 커밋 [ chore: Change member variable name ] 에 업데이트되었습니다 👍

saintym reviewed (commented):
각 코멘트 확인바랍니다!

KIRBY-PARK committed:
chore: Change member variable name
ERD와 맞춰서 멤버변수명 모두 변경했습니다.
openingHours -> openedAt
closingHours -> closedAt
minOrderAmount -> minOrderPrice

KIRBY-PARK committed:
refactor:
ShopController.java 피드백 반영
("{id}") 이 부분을 ("/{id}") 으로 변경

ShopService.java 피드백 반영
String shopName = shop.getName();
와
System.out.println("가게 '" + shopName + "' (ID: " + id + ")이 성공적으로 삭제 처리되었습니다.");
를 추가해보았는데 괜찮을지 확인 한 번 부탁드립니다 !

KIRBY-PARK committed:
Merge branch 'develop' into feature/shop-api

KIRBY-PARK committed:
Merge branch 'develop' of https://github.com/delivery-6/delivery-app into feature/shop-api

KIRBY-PARK committed:
refactor: Update ShopService to check user role for shop creation
- User 엔티티에서 isOwner 필드를 사용하지 않고, role 필드를 활용하여 가게 생성 권한을 확인하도록 수정
- ShopService 에서 role 이 'OWNER' 인 경우에만 가게를 생성할 수 있도록 권한 체크 로직 추가

TaehoHong commented on src/main/java/com/example/shop/controller/ShopController.java:
```diff
@@ -0,0 +1,55 @@
+
+    // 가게 생성
+    @PostMapping
+    public ResponseEntity<ShopReadResponseDto> create(@RequestBody ShopCreateRequestDto dto) {
+        return ResponseEntity.ok(shopService.create(dto));
+    }
+
+    // 가게 정보 업데이트
+    @PatchMapping("/{id}")
+    public ResponseEntity<ShopReadResponseDto> partialUpdate(
+            @PathVariable int id,
+            @RequestBody ShopUpdateRequestDto dto
```
유효성 검사를 할때 @RequestBody 앞에 @Valid 가 있어야 스프링에서 검사를 진행하니 
@Valid를 붙여주세요~

KIRBY-PARK replied:
넵 ! 수정 완료 했습니다 👍

KIRBY-PARK committed:
refactor : QueryDSL, Valiid Added
- 가게 정보 업데이트에 @Valid 추가
- Shop에 QueryDSL 변경 (커밋 후 재 확인 예정)

KIRBY-PARK committed:
Merge branch 'develop' of https://github.com/delivery-6/delivery-app into feature/shop-api

KIRBY-PARK committed:
redactor : Only the QueryDSL task part is rolled back
- QueryDSL을 이용하는데 서툴러서 우선 롤백하고 다시 작업해보려고 합니다.

KIRBY-PARK committed:
chore : change to QueryDSL
- QueryDSL을 이용하는 것으로 변경해보았는데 Menu쪽의 Impl도 저와 유사한 부분에서 오류가 발생하는 것을 확인했습니다.
- QShop이 자동생성되지 않아 발생하는 오류로 확인되어 함께 확인 한 번 부탁드릴게요 !

KIRBY-PARK committed:
fix : QueryDSL Error Correction
- 중복되는 코드로 인한 오류 해결

KIRBY-PARK committed:
refactor:Change List to Page
List를 사용하던 부분을 모두 Page로 변경

saintym reviewed (approved):
lgtm
